infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -292,13 +298,20 @@ private void rangeWrite(CorfuPayloadMsg<RangeWriteMsg> msg,
     }
 
     /**
-     * Resets the log unit server.
+     * Resets the log unit server via the BatchWriter.
      * Warning: Clears all data.
      */
     @ServerHandler(type = CorfuMsgType.RESET_LOGUNIT)
-    private void resetLogUnit(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
-        streamLog.reset();
-        dataCache.invalidateAll();
+    private synchronized void resetLogUnit(CorfuPayloadMsg<Long> msg,
+                                           ChannelHandlerContext ctx, IServerRouter r) {
+        // Check if the reset request is with an epoch greater than the last reset epoch seen.
+        // and should be equal to the current router epoch to prevent stale reset requests from
+        // wiping out the data.
+        if (msg.getPayload() > serverContext.getLogUnitResetEpoch()) {
+            batchWriter.reset();
+            dataCache.invalidateAll();
+            serverContext.setLogUnitResetEpoch(msg.getPayload());
+        }
```

Comment:
need to log that resetLogUnit was a noop if it falls out of the if case.

Reply:
done.